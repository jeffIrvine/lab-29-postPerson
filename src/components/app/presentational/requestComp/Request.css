.Request {
  height:40rem;
  width: 25rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: white;
  border: 3px solid black;
  border-radius: 10px;
  overflow: auto;
  max-width: 50rem;
  max-height: 50rem;
}


button{
display:inline-block;
padding:0.35em 1.2em;
border:0.1em solid #202020;
margin:0 0.3em 0.3em 0;
border-radius:0.12em;
box-sizing: border-box;
text-decoration:none;
font-family:'Roboto',sans-serif;
font-weight:300;
color:#202020;
text-align:center;
transition: all 0.2s;
}
button:hover{
color:#ffffff;
background-color:#202020;
}

