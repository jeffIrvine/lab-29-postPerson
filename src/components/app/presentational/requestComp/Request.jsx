import React from 'react';
import PropTypes from 'prop-types';

function Request({ url, method, jsonObj, onSubmit, onChange }) {
  return (
    <div>
      <form onSubmit={onSubmit} className="form-body">
        <input 
          placeholder="url" 
          type="text" 
          name="url" 
          value={url} 
          onChange={onChange} 
        />
        <div>
          <input 
            type="radio" 
            name="method" 
            value="/get" 
            checked={method === '/get'} 
            onChange={onChange}
          /> 
          <span>/get</span>

          <input 
            type="radio" 
            name="method" 
            value="/post"
            checked={method === '/post'} 
            onChange={onChange} 
          /> 
          <span>/post</span>

          <input 
            type="radio" 
            name="method" 
            value="/put"
            checked={method === '/put'} 
            onChange={onChange} 
          /> 
          <span>/put</span>

          <input 
            type="radio" 
            name="method" 
            value="/delete"
            checked={method === '/delete'} 
            onChange={onChange} 
          /> 
          <span>/delete</span>
        </div>
        <div>
          <textarea 
            name="jsonObj"
            placeholder="Input Json Object here..." 
            value={jsonObj} 
            onChange={onChange}>
          </textarea>
        </div>
        <button className="submit-button" >Submit</button>
      </form>
    </div>
  );
}

Request.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  jsonObj: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default Request;
