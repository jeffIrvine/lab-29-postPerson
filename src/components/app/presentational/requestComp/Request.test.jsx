import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Request from './Request';

describe('Request component', () => {
  afterEach(() => cleanup());
  it('renders Request', () => {
    const { asFragment } = render(<Request
      url=""
      method=""
      json="{}"
      onSubmit={ () => {} } 
      onChange={ () => {} } 
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
