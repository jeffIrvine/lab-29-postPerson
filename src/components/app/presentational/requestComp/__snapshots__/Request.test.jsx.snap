// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request component renders Request 1`] = `
<DocumentFragment>
  <div
    class="Request"
  >
    <h1>
      Welcome to postPerson!
    </h1>
    <form>
      <input
        id="urlInput"
        name="url"
        placeholder="url"
        type="text"
        value=""
      />
      <label
        for="GET"
      />
      <div>
        <label
          for="GET"
        >
          <input
            id="get"
            name="method"
            type="radio"
            value="/get"
          />
        </label>
        <span>
          /get
        </span>
        <input
          name="method"
          type="radio"
          value="/post"
        />
        <span>
          /post
        </span>
        <input
          name="method"
          type="radio"
          value="/put"
        />
        <span>
          /put
        </span>
        <input
          name="method"
          type="radio"
          value="/delete"
        />
        <span>
          /delete
        </span>
      </div>
      <div>
        <textarea
          name="jsonObj"
          placeholder="Input Json Object here..."
        />
      </div>
      <button
        class="submit-button"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
